docs/content/en/integration/openid-connect/roundcube/index.md

```diff
@@ -0,0 +1,231 @@
+* [Dovecot]
+  * [2.3.20](https://dovecot.org/doc/NEWS)
+* [Postfix]
+  * [3.7.6](https://www.postfix.org/announcements/postfix-3.8.1.html)
+* [Nginx]
+  * [1.22.1](https://nginx.org/en/CHANGES-1.22)
+
+## Before You Begin
+
+{{% oidc-common %}}
+
+### Assumptions
+
+This example makes the following assumptions:
+
+* __Application Root URL:__ `https://roundcube.example.com`
+* __Authelia Root URL:__ `https://auth.example.com`
+* __Client ID:__ `roundcube`
+* __Client Secret:__ `insecure_secret`
+
+## Configuration
+
+### Application
+
+Configure [Roundcube OAuth2] to use Authelia as an [OpenID Connect 1.0]
+Provider. Edit your [Roundcube] `/etc/roundcube/config.inc.php` configuration
+file and add the following:
+
+```php
+// Authelia doesn't accept non-localhpst HTTP redirection but RC build its
+// redirect_uri from HTTP_HOST header which is a FQDN (even if reverse-proxied
+// to localhost)
+$config['use_https'] = true;
+
+$config['oauth_provider'] = 'generic';
+$config['oauth_provider_name'] = 'Authelia OIDC';
+$config['oauth_client_id'] = roundcube';
+$config['oauth_client_secret'] = 'insecure_secret';
+$config['oauth_auth_uri'] = 'https://auth.example.com/api/oidc/authorization';
+$config['oauth_token_uri'] = 'https://auth.example.com/api/oidc/token';
+$config['oauth_identity_uri'] = 'https://auth.example.com/api/oidc/userinfo';
+$config['oauth_identity_fields'] = ['email'];
+$config['oauth_scope'] = 'email openid profile';
+// Optionally, skip Roundcube's login page
+//$config['oauth_login_redirect'] = true;
+
+// Note on IMAP and SMTP backends (cf. Debian's '/etc/roundcube/defaults.inc.php').
+// - For an IMAP instance on localhost, the default conf is enough. Otherwise,
+//   set the corresponding SSL/TLS options via 'imap_host' and 'imap_conn_options';
+// - For an SMTP instance on localhost, no auth would be required. However
+//   RC's OAuth enforces 'smtp_auth_type' = 'XOAUTH2' plus credentials (cf.
+//   <https://github.com/roundcube/roundcubemail/issues/9183>), thus you *must*
+//   use TLS or SSL via 'smtp_host' and 'smtp_conn_options'!
+```
+
+### Authelia
+
+The following YAML configuration is an example __Authelia__
+[client configuration](../../../configuration/identity-providers/openid-connect/clients.md) for use with [Roundcube]
+which will operate with the above example:
+
+```yaml
+identity_providers:
+  oidc:
+    ## The other portions of the mandatory OpenID Connect 1.0 configuration go here.
+    ## See: https://www.authelia.com/c/oidc
+    clients:
+    - id: 'roundcube'
+      description: 'Roundcube'
+      secret: '$pbkdf2-sha512$310000$c8p78n7pUMln0jzvd4aK4Q$JNRBzwAo0ek5qKn50cFzzvE9RXV88h1wJn5KGiHrD0YKtZaR/nCb2CJPOsKaPK0hjf.9yHxzQGZziziccp6Yng'  # The digest of 'insecure_secret'.
+      public: false
+      authorization_policy: 'two_factor'
+      redirect_uris:
+        - 'https://roundcube.example.com/oauth/callback/'
+      scopes:
+        - 'openid'
+        - 'profile'
+        - 'email'
```

Comment:
```suggestion
      scopes:
        - 'openid'
        - 'profile'
        - 'email'
      token_endpoint_auth_method: 'client_secret_basic'
```

Can we somehow confirm this change with the docs or the authors? It will either be client_secret_basic (header) or client_secret_post (post form)

Reply:
It should be `client_secret_post`, cf. Roundcube  OAuth [prerequisites](https://github.com/roundcube/roundcubemail/wiki/Configuration:-OAuth2#prerequisites).

Reply:
Yep LGTM! Nice find.